### PR TITLE
fix a few asan reported issues and a coredump on exit

### PR DESCRIPTION
**Pull request by gulafaran**
the wl_event_resource was running upon destruction of the compositor causing a null pointer segfault in onX11Event so ensure the event is removed upon destruction, also free the memory allocated by xcb_errors_context_new and finally call xcb_disconnect on the connection to free the fd and its memory.

also free the eventsources in hyprctl and eventloopmanager.

### CODE REVIEW

File: src/debug/HyprCtl.hpp [hunk 1]
```diff
@@ -25,6 +26,7 @@ class CHyprCtl {
     void                             startHyprCtlSocket();
 
     std::vector<SP<SHyprCtlCommand>> m_vCommands;
+    wl_event_source*                 m_eventSource{nullptr};
```
Comment (vaxerski) [inline src/debug/HyprCtl.hpp:29]: = nullptr; please

**Review by vaxerski — COMMENTED** [review]
rest lgtm

File: src/debug/HyprCtl.hpp (hunk 1 above)
Comment (gulafaran) [reply src/debug/HyprCtl.hpp]: changed. <3

**Review by vaxerski — APPROVED** [review]
thanks!

**Comment by obivan** [comment]
This one is merged, but I don't see any related commits in main. Pardon my ignorance, but... how so?

**Comment by MightyPlaza** [comment]
> This one is merged, but I don't see any related commits in main. Pardon my ignorance, but... how so?

possibly https://github.com/hyprwm/Hyprland/issues/6154#issuecomment-2144034315

**Comment by vaxerski** [comment]
thanks for spotting, I accidentally nuked a bit of history yesterday and missed restoring this one. Will do later today.

**Pull request merged**